airflow/operators/hive_operator.py

```diff
@@ -77,13 +77,16 @@ def __init__(
         self.mapred_queue_priority = mapred_queue_priority
         self.mapred_job_name = mapred_job_name
 
+        # assigned lazily
```

Comment:
Why is this needed? Might be good to explain in the comment.

Reply:
Just for consistency, as `self.hook` is an attribute assigned outside of init; also, `on_kill` assumes `self.hook` is not `None` and exists. Behavior is not changed, just making the code a bit more consistent. I'll add that in a comment.